### DISCUSSION TITLE
Tweedie on pypi

### DISCUSSION THREAD

**Discussion by wellington36** [wellington36/tweedie-numerical-approximation#4] [General]
Check https://github.com/thequackdaddy/tweedie.

```python
from scipy import integrate
import tweedie

def alpha_to_p(alpha):
    return (alpha - 2) / (alpha - 1)

for i in [0.01, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 0.99]:
    print(integrate.quad(lambda x: tweedie.tweedie(mu=1, phi=1, p=alpha_to_p(i)).pdf(x), 1e-6, 50, points=1000))
```

Out:

|alpha | points | I                   | absolute error           |
|---------|-----------|-----------------|-----------------|
|0.01   | 1000     |Nan| Nan|
|0.1     | 1000     |1.0000e+00| 1.4701e-04|
|0.2     | 1000     |0.9950e+00| 1.1414e-04|
|0.3     | 1000     |0.9929e+00| 5.3928e-03|
|0.4     | 1000     |0.9758e+00| 1.2567e-02|
|0.5     | 1000     |1.0000e+00| 3.3829e-09|
|0.6     | 1000     |Nan| Nan|
|0.7     | 1000     |$\infty$| $\infty$|
|0.8     | 1000     |$\infty$| $\infty$|
|0.9     | 1000     |0.0000e+00| 0.0000e-00|
|0.99   | 1000     |Nan| Nan|

**Edit:** updated code

**Comment by maxbiostat** [comment]
Seems stable enough, eh? I wonder what the catch is... :thinking:

**Comment by wellington36** [comment]
1. ~~Look at the column of errors; they are all the same, which seems suspicious to me. Not only are those decimal places in the next 10 positions all the same as well.~~

2. In their code, they don't have many references or proof that it works. In an issue, they commented on a function that calculates the cumulative distribution function (CDF), saying, 'I believe I was able to validate the results against the R package. At low rates, it should be fine. Only at high rates could I see a problem.' (Source: [GitHub issue](https://github.com/thequackdaddy/tweedie/issues/9))

3. Their focus is on 1 < p < 2. They state, 'The main focus of this package is the compound-Poisson behavior, specifically where 1 < p < 2. However, it should be possible to calculate the distribution for all possible values of p.' Our focus, however, is on $2 < p < \infty$ because if $0 < \alpha = \frac{p-2}{p-1}< 1$, then $2 < p < \infty$. This is different from their approach.

In conclusion, I think one must carefully examine what is being done.

**Edit 1:** Problem (1) solved.

**Comment by wellington36** [comment]
@maxbiostat I find an error, I forgot to update the alpha in the obvious code that (1) happens. Now I'm more convinced that code doesn't make miracles. It's closer to the naive version.
